core/src/main/java/org/fao/geonet/kernel/batchedit/CSVBatchEdit.java

```diff
@@ -0,0 +1,377 @@
+				}
+			} else {
+				try {
+					Element element = (Element) _xpath.selectSingleNode(metadata);
+					if(element != null){
+						element.setText(csvr.get(headerVal));
+					}
+					else{
+						BatchEditParam e = new BatchEditParam(_xpath.getXPath(),
+								"<gn_add>" + csvr.get(headerVal) + "</gn_add>");
+						listOfUpdates.add(e);
+					}
+				} catch (Exception e) {
+					//Log.error(Geonet.SEARCH_ENGINE, "Unable to set text for eCatId/UUID " + id +" for the value " + csvr.get(headerVal) +", " + e.getLocalizedMessage());
+					report.getErrorInfo().add("Unable to set text for the value, " + headerVal + ": "+ csvr.get(headerVal) +", " + e.getMessage());
+				}	
+			}
+		}
+		
+		return report;
+	}
+
+	
+	@Override
+	public void setApplicationEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
+		this.applicationEventPublisher = applicationEventPublisher;
+	}
+	
+	/**
+	 * If one of the element is not defined in CSV, it won't remove both element. It only adds the element.
+	 * 
+	 * @param header
+	 * @param csvr
+	 * @return
+	 */
+	private boolean checkDependencies(String header, CSVRecord csvr){
+		
+		if(Geonet.EditType.GEOBOX.equals(header)){
+			if(!csvr.isMapped(Geonet.EditType.VERTICAL)){
+				return false;
+			}
+		}
+		
+		if(Geonet.EditType.VERTICAL.equals(header)){
+			if(!csvr.isMapped(Geonet.EditType.GEOBOX)){
+				return false;
+			}
+		}
+		
+		return true;
+	}
+
+
+	/**
+	 * Search metadata using lucene index
+	 * @param context
+	 * @param srvContext
+	 * @param request
+	 * @return
+	 * @throws BatchEditException
+	 */
+	public Metadata getMetadataByLuceneSearch(ApplicationContext context, ServiceContext srvContext, Element request) throws BatchEditException {
+
+		LuceneQueryInput luceneQueryInput = new LuceneQueryInput(request);
+		
+		Query _query = new LuceneQueryBuilder(luceneConfig, luceneConfig.getTokenizedField(),
+				SearchManager.getAnalyzer(Geonet.DEFAULT_LANGUAGE, true), Geonet.DEFAULT_LANGUAGE)
+						.build(luceneQueryInput);
+
+		MetadataRepository mdRepo = context.getBean(MetadataRepository.class);
+
+		try {
+			
+			TopDocs tdocs = searcher.search(_query, 1);
+			DocumentStoredFieldVisitor docVisitor = new DocumentStoredFieldVisitor("_uuid");
+
+			indexAndTaxonomy.indexReader.document(tdocs.scoreDocs[0].doc, docVisitor);
+			org.apache.lucene.document.Document doc = docVisitor.getDocument();
+
+			String uuid = doc.get("_uuid");
+			Log.debug(Geonet.SEARCH_ENGINE, "getMetadataByLuceneSearch --> uuid: " + uuid);
+
+			if (uuid != null) {
+				Metadata md = mdRepo.findOneByUuid(uuid);
+				return md;
+			}
+
+		} catch (Exception e) {
+			throw new BatchEditException("failed to get this record using lucene search");
+		}
+
+		return null;
+	}
+	
+	/**
+	 * Retrieves CRS reference system
+	 * @param crsId
+	 * @return
+	 */
+	public Crs getById(String crsId) {
```

Comment:
The `Crs` class returned is interesting, I am used to dealing with `CoordinateReferenceSystem` objects from GeoTools.

The lookup portion of the method appears to be a reproduction of the `CRS.decode( crsId )` utility method:
- slow: it seems to traverse all of the codes one by one
- incomplete: the format `<Factory>:<code>` has known limitations, for example it is subject to axis order issue where EPSG database axis order and common practice diverge

Reference:

* [CRS](http://docs.geotools.org/latest/userguide/library/referencing/crs.html) (GeoTools User Guide)
* [Axis Order](http://docs.geotools.org/latest/userguide/library/referencing/order.html)  (GeoTools User Guide)

Reply:
Digging deeper I can see the `Crs` class which simplifies the GeoTools CoordinateReferenceSystem object to a non-functional description.

This method appears to be a duplication of `CrsUtils.getById(String crsId)` - so the real question is why is this functionality duplicated. And then a recommendation to produce an cache rather than traverse the EPSG database each time.